src/Downloader.ts

```diff
@@ -649,15 +649,8 @@ class Downloader {
 
     try {
       if (this.optimisationCacheUrl && this.isImageUrl(requestOptions.url)) {
-        this.s3.downloadIfPossible(this.stripHttpFromUrl(requestOptions.url), requestOptions.url).then(async (s3ImageResp) => {
-          if (s3ImageResp) {
-            handler(null, {
-              responseHeaders: s3ImageResp.headers,
-              content: s3ImageResp.imgData,
-            });
-          } else {
-            await this.imageDownloadCompressAndUploadToS3(requestOptions, handler);
-          }
+        this.s3.downloadIfPossible(this.stripHttpFromUrl(requestOptions.url)).then(async (s3ImageResp) => {
+          this.checkStatusAndUploadtoS3(requestOptions, handler, s3ImageResp);
```

Comment:
Merge this two lines and with `checkStatusAndUploadtoS3()` and call the method `downloadImage()`.